study_preferences.py

```diff
@@ -113,7 +102,7 @@ def __update_form_data_add_argument(self, update_form_data) -> None:
     # a private method to validate the user input data before inserting them into the database
     def __validate_form_data(self, errors, **kwargs) -> None:
         for key, value in kwargs.items():
-            if key == 'courses_preferences':
+            if key == 'courses_preferences' and value:
```

Comment:
inconsistent format, just run 

`black study_preferences.py` in the command line to fix this